src/app/core/services/show-style-variant-state.service.ts

```diff
@@ -3,13 +3,40 @@ import { BehaviorSubject, lastValueFrom, Observable } from 'rxjs'
 import { ShowStyleVariant } from '../models/show-style-variant'
 import { ShowStyleVariantService } from '../abstractions/show-style-variant.service'
 import { EventSubscription } from '../../event-system/abstractions/event-observer.service'
+import { RundownEventObserver } from './rundown-event-observer.service'
+import { RundownUpdatedEvent } from '../models/rundown-event'
+import {Logger} from "../abstractions/logger.service";
 
 @Injectable()
 export class ShowStyleVariantStateService implements OnDestroy {
   private readonly showStyleVariantSubjects: Map<string, BehaviorSubject<ShowStyleVariant>> = new Map()
   private readonly eventSubscriptions: EventSubscription[]
+  private readonly logger: Logger
 
-  constructor(private readonly showStyleVariantService: ShowStyleVariantService) {}
+  constructor(
+    private readonly showStyleVariantService: ShowStyleVariantService,
+    private readonly rundownEventObserver: RundownEventObserver,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('ShowStyleVariantStateService')
+    this.subscribeToRundownEvents()
+  }
+
+  private subscribeToRundownEvents(): void {
+    this.rundownEventObserver.subscribeToRundownUpdates(this.updateShowStyleVariantFromEvent.bind(this))
```

Comment:
This should be assigned to the `eventSubscriptions`, such that we can unsubscribe on destroy.

Reply:
This is not going to be part of this PR, but it would be nice if we could refactor away from returning `EventSupbscription` so that we could simply do `.pipe(takeUntil(this.destroySubject))`

Reply:
For now it has been added to `eventSubscriptions`.